pkg/cloudprovider/providers/azure/azure.go

```diff
@@ -0,0 +1,143 @@
+package azure
+
+import (
+	"encoding/json"
+	"io"
+
+	"k8s.io/kubernetes/pkg/cloudprovider"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/glog"
+)
+
+const AzureProviderName = "azure"
+
+type AzureConfig struct {
+	Cloud             string `json:"cloud"`
```

Comment:
Actually, it looks like both https://github.com/ghodss/yaml and http://godoc.org/gopkg.in/yaml.v2 are in Godeps, but ghodss flavor is used more (not a scientific study). It also seems to translate yaml->json and then decodes into the struct, obviating the need for yaml tags, but I suppose it can't hurt and I'll add tests either way.